edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsSerializer.cs

```diff
@@ -22,13 +23,13 @@ public static IEnumerable<byte> MetricsToBytes(IEnumerable<Metric> metrics)
                 x.Select(m => new RawMetricValue(m.TimeGeneratedUtc, m.Value))));
         }
 
-        static IEnumerable<byte> MetricGroupsToBytes(string name, string tags, IEnumerable<RawMetricValue> rawValues)
+        static IEnumerable<byte> MetricGroupsToBytes(string name, Dictionary<string, string> tags, IEnumerable<RawMetricValue> rawValues)
         {
             byte[] nameArray = Encoding.UTF8.GetBytes(name);
             byte[] nameLength = BitConverter.GetBytes(name.Length);
 
-            byte[] tagsArray = Encoding.UTF8.GetBytes(tags);
-            byte[] tagsLength = BitConverter.GetBytes(tags.Length);
+            byte[] tagsArray = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(tags));
```

Comment:
I am not sure SerializeObject will always serialize dictionary items in the same order; please check.

Reply:
Since equals will no longer check for order, does this still matter?